cmake/HPX_SetupAllocator.cmake

```diff
@@ -34,8 +36,12 @@ if(NOT HPX_WITH_MALLOC_DEFAULT)
       hpx_error(${allocator_error})
     endif()
 
-    set_property(TARGET hpx::allocator PROPERTY
-      INTERFACE_LINK_LIBRARIES ${TCMALLOC_LIBRARIES})
+    if(${CMAKE_VERSION} VERSION_LESS "3.12.0")
```

Comment:
This was just an educated guess on my part. @aurianer do you remember at which version of CMake `target_link_libraries(target INTERFACE ...)` started working?

Reply:
I think it is 3.10 but the problem is `target_link_libraries` for `INTERFACE` `IMPORTED` libraries in particular

Reply:
I think we can leave this as is as anybody affected by this problem can simply update cmake to the latest version. But if you think it's worth doing I can certainly apply a change making this compare agains "3.10.0" instead.

Reply:
That's fine as it is for me :)